modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -58,52 +58,67 @@ object ConsensusStateUpdater {
       key: Key,
       lastKeyAndArtifact: (Key, Artifact)
     ): F[Option[ConsensusState[Key, Artifact]]] =
       consensusStorage
-        .updateState(key)(logStatusIfUpdated(internalTryFacilitateConsensus(key, lastKeyAndArtifact)))
+        .condModifyState(key)(toModifyStateFn(internalTryFacilitateConsensus(key, lastKeyAndArtifact)))
+        .flatMap(evalEffect)
+        .flatTap(logIfModified(key))
 
     def tryAdvanceConsensus(
       key: Key,
       resources: ConsensusResources[Artifact]
     ): F[Option[ConsensusState[Key, Artifact]]] =
       consensusStorage
-        .updateState(key)(logStatusIfUpdated(internalTryAdvanceConsensus(key, resources)))
-
-    import consensusStorage.StateUpdateFn
-
-    private def logStatusIfUpdated(fn: StateUpdateFn): StateUpdateFn =
-      fn(_, _).flatTap {
-        case Some((true, newState)) =>
-          logger.debug(s"Consensus status for key ${newState.key.show} transitioned to ${newState.status.show}")
-        case _ => Applicative[F].unit
+        .condModifyState(key)(toModifyStateFn(internalTryAdvanceConsensus(key, resources)))
+        .flatMap(evalEffect)
+        .flatTap(logIfModified(key))
+
+    import consensusStorage.ModifyStateFn
+
+    private def toModifyStateFn(
+      fn: Option[ConsensusState[Key, Artifact]] => F[Option[(ConsensusState[Key, Artifact], F[Unit])]]
+    ): ModifyStateFn[ModifyStateResult] =
+      maybeState =>
+        fn(maybeState).map(_.map {
+          case (state, effect) => (state.some, (state.some, effect))
+        })
+
+    private def evalEffect(result: Option[ModifyStateResult]): F[Option[ConsensusState[Key, Artifact]]] =
+      result.flatTraverse {
+        case (maybeState, effect) => effect >> maybeState.pure[F]
       }
 
+    private def logIfModified(key: Key)(maybeState: Option[ConsensusState[Key, Artifact]]): F[Unit] =
+      maybeState.traverse { state =>
+        logger.debug(s"Consensus status for key ${key.show} transitioned to ${state.status.show}")
+      }.void
+
     private def internalTryFacilitateConsensus(key: Key, lastKeyAndArtifact: (Key, Artifact))(
-      maybeState: Option[ConsensusState[Key, Artifact]],
-      setter: Option[ConsensusState[Key, Artifact]] => F[Boolean]
-    ): F[Option[(Boolean, ConsensusState[Key, Artifact])]] =
+      maybeState: Option[ConsensusState[Key, Artifact]]
+    ): F[Option[(ConsensusState[Key, Artifact], F[Unit])]] =
       maybeState match {
         case None =>
           for {
             peers <- clusterStorage.getPeers
             facilitators = selfId :: NodeState.ready(peers).map(_.id).toList
             state = ConsensusState(key, facilitators, lastKeyAndArtifact, Facilitated[Artifact]())
-            result <- setter(state.some)
             bound <- consensusStorage.getUpperBound
-            _ <- gossip.spread(ConsensusFacility(key, bound))
-          } yield (result, state).some
+            effect = gossip.spread(ConsensusFacility(key, bound))
```

Comment:
`gossip.spread` raise errors sometimes (see above comment about error handling)

Reply:
As stated above, there's very little we can do when evaluating an effect after the state transition. Once the health-check is in place the node will get kicked out anyway.